src/ray/core_worker/reference_count.cc

```diff
@@ -628,18 +637,25 @@ bool ReferenceCounter::SetDeleteCallback(
   return true;
 }
 
-std::vector<ObjectID> ReferenceCounter::ResetObjectsOnRemovedNode(
-    const NodeID &raylet_id) {
+void ReferenceCounter::ResetObjectsOnRemovedNode(const NodeID &raylet_id) {
   absl::MutexLock lock(&mutex_);
-  std::vector<ObjectID> lost_objects;
   for (auto it = object_id_refs_.begin(); it != object_id_refs_.end(); it++) {
     const auto &object_id = it->first;
-    if (it->second.pinned_at_raylet_id.value_or(NodeID::Nil()) == raylet_id) {
-      lost_objects.push_back(object_id);
+    if (it->second.pinned_at_raylet_id.value_or(NodeID::Nil()) == raylet_id ||
+        it->second.spilled_node_id == raylet_id) {
```

Comment:
Isn't `it->second.spilled_node_id` always equivalent to `it->second.pinned_at_raylet_id`? Is this if predicate necessary? (If they are not the same, it could be a bug because we release a pinned object when the plasma object is spilled?)

Reply:
Hmm I think you're right but I think this is a bit easier to understand.

Reply:
Maybe I prefer `RAY_CHECK(it->second.spilled_node_id == raylet_id)` (in case this invariant changes), but it is not a strong requirements.